yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -296,6 +284,53 @@ private[spark] class Client(
     }
 
     /**
+     * Distribute a file to the cluster.
+     *
+     * @param path URI of the file to distribute.
+     * @param resType Type of resource being distributed.
+     * @param destName Name of the file in the distributed cache.
+     * @param targetDir Subdirectory where to place the file.
+     * @param appMasterOnly Whether to distribute only to the AM.
+     * @return A 2-tuple. First item is whether the file is a "local:" URI. Second item is the
+     *         localized path for non-local paths, or the input `path` for local paths.
```

Comment:
The second item can be null. Can you document when this is the case to avoid NPEs in the future if this isn't used carefully?